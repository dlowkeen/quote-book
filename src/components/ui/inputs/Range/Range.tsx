import React from 'react';

import * as styles from '../inputs.css';

interface IRangeProps {
  label: string;
  value: string;
  onChange: (e: any) => void;
  max: string;
}

const Range: React.FunctionComponent<IRangeProps> = ({
  label,
  value,
  onChange,
  max,
}) => (
  <div className={`${styles.formElement} ${styles.formElementRange}`}>
    <label>{label}</label>
    <input
      type='range'
      value={value}
      onChange={onChange}
      max={max}
      min='0'
      step='1'
    />
  </div>
);

export default Range;
