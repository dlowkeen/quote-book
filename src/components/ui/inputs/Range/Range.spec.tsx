import React from 'react';
import renderer from 'react-test-renderer';

import Range from './Range';

describe('The Range component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <Range max='10' value='3' label='label' onChange={() => 'changed!'} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
