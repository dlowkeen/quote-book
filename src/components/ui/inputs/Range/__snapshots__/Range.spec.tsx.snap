// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Range component renders as expected 1`] = `
<div
  className="inputs__form-element__VT4Ts inputs__form-element-range__2ZDm_"
>
  <label>
    label
  </label>
  <input
    max="10"
    min="0"
    onChange={[Function]}
    step="1"
    type="range"
    value="3"
  />
</div>
`;
